.github/workflows/build.yml

```diff
@@ -18,20 +21,50 @@ jobs:
         operating-system: [ubuntu-latest, windows-latest, macOS-latest]
 
     steps:
+      - run: echo "matrix.os - ${{ matrix.operating-system }}"
       - name: Checkout
         uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
 
       - name: Setup Node.js 16
         uses: actions/setup-node@v3
         with:
           node-version: 16.x
 
-      - name: Build
+      - name: Recompile and Format action
         run: npm run build
 
-      - name: Format
-        run: npm run format
+      # These last steps only need to be done once
+      # per commit.  Run them on the ubuntu job.
```

Comment:
I may have missed something, but the steps below don't look like they reference anything from the steps above. So out of curiosity, could they be put into a different job that relies on the success of this matrix job? Or is that more effort/confusing/etc.?

Reply:
The commit step would depend on the recompile and format step (which happens on all three matrix jobs) if there are outstanding changes from there.

Reply:
Oh right, good point. I guess we could pull out ubuntu from the matrix and make it a separate job, but that would look a little weird.